File: solutions/387.first-unique-character-in-a-string/first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
function firstUniqChar (s) {
  for (let i = 0; i < s.length; i++) {
    let item = s[i]
    if (s.indexOf(item) === s.lastIndexOf(item)) return i
  }

  return -1
}

File: solutions/387.first-unique-character-in-a-string/first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
	hash := map[string]int{}
	for _, c := range s {
		char := string(c)
		hash[char]++
	}

	for i, c := range s {
		char := string(c)
		if hash[char] == 1 {
			return i
		}
	}
	return -1
}
